base/multi.jl

```diff
@@ -1071,44 +1183,44 @@ function process_hdr(s, validate_cookie)
     return VersionNumber(strip(String(version)))
 end
 
-function handle_msg(msg::CallMsg{:call}, r_stream, w_stream, version)
-    schedule_call(msg.response_oid, ()->msg.f(msg.args...; msg.kwargs...))
+function handle_msg(msg::CallMsg{:call}, msghdr, r_stream, w_stream, version)
+    schedule_call(msghdr.response_oid, ()->msg.f(msg.args...; msg.kwargs...))
 end
-function handle_msg(msg::CallMsg{:call_fetch}, r_stream, w_stream, version)
+function handle_msg(msg::CallMsg{:call_fetch}, msghdr, r_stream, w_stream, version)
     @schedule begin
         v = run_work_thunk(()->msg.f(msg.args...; msg.kwargs...), false)
-        deliver_result(w_stream, :call_fetch, msg.response_oid, v)
+        deliver_result(w_stream, :call_fetch, msghdr.response_oid, v)
     end
 end
 
-function handle_msg(msg::CallWaitMsg, r_stream, w_stream, version)
+function handle_msg(msg::CallWaitMsg, msghdr, r_stream, w_stream, version)
     @schedule begin
-        rv = schedule_call(msg.response_oid, ()->msg.f(msg.args...; msg.kwargs...))
-        deliver_result(w_stream, :call_wait, msg.notify_oid, fetch(rv.c))
+        rv = schedule_call(msghdr.response_oid, ()->msg.f(msg.args...; msg.kwargs...))
+        deliver_result(w_stream, :call_wait, msghdr.notify_oid, fetch(rv.c))
     end
 end
 
-function handle_msg(msg::RemoteDoMsg, r_stream, w_stream, version)
+function handle_msg(msg::RemoteDoMsg, msghdr, r_stream, w_stream, version)
     @schedule run_work_thunk(()->msg.f(msg.args...; msg.kwargs...), true)
 end
 
-function handle_msg(msg::ResultMsg, r_stream, w_stream, version)
-    put!(lookup_ref(msg.response_oid), msg.value)
+function handle_msg(msg::ResultMsg, msghdr, r_stream, w_stream, version)
+    put!(lookup_ref(msghdr.response_oid), msg.value)
 end
 
-function handle_msg(msg::IdentifySocketMsg, r_stream, w_stream, version)
+function handle_msg(msg::IdentifySocketMsg, msghdr, r_stream, w_stream, version)
     # register a new peer worker connection
     w=Worker(msg.from_pid, r_stream, w_stream, cluster_manager; version=version)
     send_connection_hdr(w, false)
-    send_msg_now(w, IdentifySocketAckMsg())
+    send_msg_now(w, MsgHeader(), IdentifySocketAckMsg())
```

Comment:
While it can incidentally reuse `msghdr` sent as part of `IdentifySocketMsg` (since that is empty too), it is clearer to specify the header to send back as part of `IdentifySocketAckMsg` separately.